src/extension/decorations/closing_labels_decorations.ts

```diff
@@ -57,27 +65,31 @@ export class ClosingLabelsDecorations implements vs.Disposable {
 			// Get the end of the line where we'll show the labels.
 			const endOfLine = this.activeEditor.document.lineAt(finalCharacterPosition).range.end;
 
+			const closingLabelsPrefix = config.closingLabelsPrefix;
+
 			const existingDecorationForLine = decorations[endOfLine.line];
 			if (existingDecorationForLine) {
-				existingDecorationForLine.renderOptions.after.contentText = " // " + r.label + " " + existingDecorationForLine.renderOptions.after.contentText;
+				existingDecorationForLine.renderOptions.after.contentText = closingLabelsPrefix + r.label + " " + existingDecorationForLine.renderOptions.after.contentText;
 			} else {
 				const dec = {
 					range: new vs.Range(this.activeEditor.document.positionAt(r.offset), endOfLine),
-					renderOptions: { after: { contentText: " // " + r.label } },
+					renderOptions: { after: { contentText: closingLabelsPrefix + r.label } },
 				};
 				decorations[endOfLine.line] = dec;
 			}
 		}
 
-		this.activeEditor.setDecorations(this.decorationType, Object.keys(decorations).map((k) => parseInt(k, 10)).map((k) => decorations[k]));
+		this.recreateDecorationType();
```

Comment:
Recreating this every time is probably overkill (I don't know what work VS Code might end up doing if we keep disposing and creating new kinds), but we could set up a handler in the constructor that listens to config changes similar to we do for formatting:

```ts
		this.disposables.push(workspace.onDidChangeConfiguration((e) => {
			if (e.affectsConfiguration("dart.enableSdkFormatter")) {
				if (config.enableSdkFormatter)
					this.registerAllFormatters();
				else
					this.unregisterAllFormatters();
			}
		}));
```